commit f6f34fd0cd6d523cc472351bcdc9b9ae180aac41
Author: Sven Neumann <sven@gimp.org>
Date:   Mon Nov 3 16:39:20 2008 +0000

    patch: jpeg-units
    
    Bug 559081 – JPEG Save dialog preview should adjust size units
    
    2008-11-03  Sven Neumann  <sven@gimp.org>
    
    	Bug 559081 – JPEG Save dialog preview should adjust size units
    
    	* plug-ins/file-jpeg/jpeg-save.c: use
    g_format_size_for_display()
    	to display the JPEG file size.
    
    svn path=/trunk/; revision=27532
    
    Signed-off-by: Nils Philippsen <nils@redhat.com>

diff --git a/plug-ins/file-jpeg/jpeg-save.c b/plug-ins/file-jpeg/jpeg-save.c
index 2d0d249..35cda17 100644
--- a/plug-ins/file-jpeg/jpeg-save.c
+++ b/plug-ins/file-jpeg/jpeg-save.c
@@ -192,14 +192,14 @@ background_jpeg_save (PreviewPersistent *pp)
       /* display the preview stuff */
       if (!pp->abort_me)
         {
-          struct stat buf;
-          gchar       temp[128];
+          struct stat  buf;
+          gchar       *text;
 
           g_stat (pp->file_name, &buf);
-          g_snprintf (temp, sizeof (temp),
-                      _("File size: %02.01f kB"),
-                      (gdouble) (buf.st_size) / 1024.0);
-          gtk_label_set_text (GTK_LABEL (preview_size), temp);
+          text = g_strdup_printf (_("File size: %s"),
+				  g_format_size_for_display (buf.st_size));
+          gtk_label_set_text (GTK_LABEL (preview_size), text);
+	  g_free (text);
 
           /* and load the preview */
           load_image (pp->file_name, GIMP_RUN_NONINTERACTIVE, TRUE, NULL);
